public/app/features/alerting/unified/components/receivers/form/fields/OptionField.tsx

```diff
@@ -75,33 +83,34 @@ export const OptionField: FC<Props> = ({
         pathPrefix={optionPath}
         readOnly={readOnly}
         pathIndex={pathPrefix}
+        parentOption={parentOption}
         customValidator={customValidator}
-        isSecure={isSecure}
+        onResetSecureField={onResetSecureField}
+        secureFields={secureFields}
       />
     </Field>
   );
 };
 
-const OptionInput: FC<Props & { id: string; pathIndex?: string; isSecure?: boolean }> = ({
+const OptionInput: FC<Props & { id: string; pathIndex?: string }> = ({
   option,
   invalid,
   id,
   pathPrefix = '',
   pathIndex = '',
   readOnly = false,
   customValidator,
-  isSecure,
+  onResetSecureField,
+  secureFields = {},
+  parentOption,
 }) => {
   const styles = useStyles2(getStyles);
   const { control, register, unregister, getValues, setValue } = useFormContext();
+
   const name = `${pathPrefix}${option.propertyName}`;
+  const nestedKey = parentOption ? `${parentOption.propertyName}.${option.propertyName}` : option.propertyName;
```

Comment:
If there is no parent, then the key is not _nested_?

Reply:
Is there any relation between `pathPrefix` and `parentOption`?

Reply:
I would say that the `pathPrefix` is related to the form and `parentOption` is more about this particular field.But I'm not sure, honestly.